### PR TITLE
Start integrated tests for basic questions

### PR DESCRIPTION
**Pull request by plocket**
Sorry it's a bunch of stuff. I didn't realize how far I'd gotten without making a PR. I hope most of it is straightforward.

Issue to add - adapt more `click`s to be flexible so they can be on mobile too.

### CODE REVIEW

File: tests/features/support/steps.js [hunk 1]
```diff
@@ -4,30 +4,89 @@ const puppeteer = require('puppeteer');
...
 
+let device_touch_map = {
+  mobile: 'tap',
+  pc: 'click',
+};
+
 // -- Puppeteer specific steps from hello_world.feature
 
-Given(/I start the interview/, async () => {
+Given(/I start the interview ?(.*)/, async (optional_device) => {
```
Comment (niharikasingh) [inline tests/features/support/steps.js:49]: Reword this to something like `I start the interview on a [phone/pc]`? That way it sounds more like natural language.

File: tests/features/support/steps.js [hunk 2]
```diff
@@ -51,38 +110,158 @@ When(/I click the (button|link) "([^"]+)"/, async (elemType, phrase) => {
...
 });
 
+Then('I should see the link {string}', async (linkText) => {
+  let [link] = await scope.page.$x(`//a[contains(text(), "${linkText}")]`);
+  expect(link).to.exist;
+});
+
+// TODO: Research how da handles invalid symbols
+Then('the question id should be {string}', async (question_class) => {
+  /* Looks for the question id as it's written in the .yml.
```
Comment (niharikasingh) [inline tests/features/support/steps.js:144]: Is this looking for a question ID or question class? The sentence says ID but the `waitFor` function below is looking for a class.

File: tests/features/support/steps.js [hunk 3]
```diff
@@ -51,38 +110,158 @@ When(/I click the (button|link) "([^"]+)"/, async (elemType, phrase) => {
...
+  let response = await linkPage.goto(actual_url, {waitUntil: 'domcontentloaded'});
+  expect( response.ok() ).to.be.true;
+  linkPage.close()
+});
+
+Then(/the link "([^"]+)" should open in (a new window|the same window)/, async (linkText, which_window) => {
+  let [link] = await scope.page.$x(`//a[contains(text(), "${linkText}")]`);
+
+  let prop_obj = await link.getProperty('target');
+  let target = await await prop_obj.jsonValue();
```
Comment (niharikasingh) [inline tests/features/support/steps.js:187]: Should this be clicking on the link? Right now, I think it's just checking whether the link opens in a new window or not.

File: tests/features/support/scope.js [hunk 4]
```diff
@@ -1 +1,59 @@
...
+      let elems_array = Array.from( elements );
+      for ( let elem of elems_array ) {
+        if (( elem.innerText ).includes( field_label )) {
+          return elem.getAttribute( 'for' );
+        }
+      }  // End for labels
+
+    }, field_label);
+
+    return field_id;
```
Comment (niharikasingh) [inline tests/features/support/scope.js:57]: I'm not sure these functions belong in `scope`. This file is just to make sure that the `page` and `browser` variables can be shared across the various test helper functions. These look like helper functions for the tests, and so they would fit into `tests/features/support/steps.js` or another file entirely.

**Review by niharikasingh — APPROVED** [review]

File: tests/features/support/steps.js (hunk 1 above)
Comment (plocket) [reply tests/features/support/steps.js:49]: Do you mean that the function would give more information about what it's looking for? I suspect a lot of people using this may not know how to look for that information, but maybe it'd be good to be more specific and have them there anyway. Can that be a future issue?

Note: `(pc|mobile)` may be the best bet as phone or tablet are both in that category.

File: tests/features/support/steps.js (hunk 2 above)
Comment (plocket) [reply tests/features/support/steps.js:144]: This refers to the docassemble `question` and `id`:
```
id: user name
question: |
  What is your name?
# And so on
```

These cucumber scripts are meant to be written mostly by people that are familiar with docassemble. Many of the folks I've been working with don't know html, javascript, how to examine the DOM, etc.

[Edit: People editing docassemble code will be able to see it, but MassAccess interviews also show their ids at the top of the screen when folks are on the developer servers, so they don't even have to edit the YAML to know what the id is.]

Comment (plocket) [reply tests/features/support/steps.js:144]: Something that does need to be a future issue is how to sanitize the id text in the same way that docassemble does it so that special characters can be handled. I think I have a note to that effect.

File: tests/features/support/steps.js (hunk 3 above)
Comment (plocket) [reply tests/features/support/steps.js:187]: There was some complication involved in clicking on the link. I don't remember what. Maybe it was that some buttons, like the escape button, won't open a new window - just change the current one, so I didn't want links in general to actually be clicked on. Maybe it was something else? Maybe I had trouble detecting a new window opening with puppeteer?

To sum up, it is deliberately not clicking on the link, though it might be worth discussing alternatives. Perhaps for a future issue? (I only mention making issues because if this is pulled in, other development can happen at the same time as this is being improved.)

File: tests/features/support/scope.js (hunk 4 above)
Comment (plocket) [reply tests/features/support/scope.js:57]: Functions that are just in `tests/features/support/steps.js` are not accessible to the actual steps. I don't think they keep that context somehow. At least, it didn't work when I tried it, but maybe I did something wrong. For that reason, I had suspected it wouldn't work in other files, though I didn't try.

I think I also saw some examples out there that used `scope` for this kind of thing, but maybe they were bad examples. We can make this a research issue. If there's an established practice out there, I'm interested in learning more about it.

File: tests/features/support/steps.js (hunk 3 above)
Comment (plocket) [reply tests/features/support/steps.js:187]: I realize you've approved, but I want to be clear that I'm not dismissing these. They've all been good to bring up.

File: tests/features/support/steps.js (hunk 1 above)
Comment (plocket) [reply tests/features/support/steps.js:49]: Oh, I remember now - I was also trying to be flexible in these ways:
1. Allow the person to just say 'I start the interview' and not have to specify a device.
2. All flexibility for words in between, like allowing 'a' or 'the' or 'an' ('an iphone').

Again, worth discussing whether that flexibility will add confusion.

File: tests/features/support/steps.js (hunk 2 above)
Comment (plocket) [reply tests/features/support/steps.js:144]: If we do add looking for actual html ids, my plan is to use the text, 'The element with the id {string} is present' or something.
